rust/frontend/src/function.rs

```diff
@@ -0,0 +1,504 @@
+//! This module provides idiomatic Rust API for creating and working with TVM functions.
+//!
+//! For calling an already registered TVM function use [`function::Builder`] and to register
+//! a TVM packed function from Rust either use [`function::register`] or the
+//! macro [`register_global_func`].
+//!
+//! See the `tests` and `examples` repository for usage examples.
+
+use std::{
+    ffi::{CStr, CString},
+    mem,
+    os::raw::{c_char, c_int, c_void},
+    ptr, slice, str,
+    sync::Mutex,
+};
+
+use ts;
+
+use ty::TypeCode;
+use value::{TVMValue, ValueKind};
+use ErrorKind;
+use Module;
+use Result;
+use TVMArgValue;
+use TVMRetValue;
+
+lazy_static! {
+    static ref GLOBAL_FUNCTION_NAMES: Mutex<Vec<&'static str>> = {
+        let mut out_size = 0 as c_int;
+        let name = ptr::null_mut() as *mut c_char;
+        let mut out_array = name as *mut _;
+        check_call!(ts::TVMFuncListGlobalNames(
+            &mut out_size as *mut _,
+            &mut out_array
+        ));
+        let names_list = unsafe { slice::from_raw_parts(out_array, out_size as usize) };
+        Mutex::new(
+            names_list
+                .into_iter()
+                .map(|&p| unsafe { CStr::from_ptr(p).to_str().unwrap() })
+                .collect(),
+        )
+    };
+}
+
+/// Returns a registered TVM function by name.
+pub fn get_global_func(name: &str, is_global: bool, allow_missing: bool) -> Option<Function> {
```

Comment:
rustc should be complaining about `allow_missing` being unused. you should feel free to remove it!

Reply:
Ah, right! I was suppressing it in `unused_variables`. Will clean it up.